File: nas_spec/code_samples/Python/workflows@{workflowId}@conditions@{workflowConditionId}/put.py
# For more information please refer to https://github.com/checkout/checkout-sdk-python
import checkout_sdk
from checkout_sdk.environment import Environment
from checkout_sdk.exception import CheckoutApiException, CheckoutArgumentException, CheckoutAuthorizationException
from checkout_sdk.four.oauth_scopes import OAuthScopes
from checkout_sdk.workflows.workflows import EventWorkflowConditionRequest

# API Keys
api = checkout_sdk.FourSdk() \\
    .secret_key('secret_key') \\
    .public_key('public_key') \\
    .environment(Environment.sandbox()) \\
    .build()
# or Environment.production()

# OAuth
api = checkout_sdk.OAuthSdk() \\
    .client_credentials('client_id', 'client_secret') \\
    .environment(Environment.sandbox()) \\
    .scopes([OAuthScopes.FLOW, OAuthScopes.FLOW_WORKFLOWS, OAuthScopes.FLOW_EVENTS]) \\
    .build()

request = EventWorkflowConditionRequest()
request.events = {'gateway': ['card_verified',
                              'card_verification_declined',
                              'payment_approved',
                              'payment_pending',
                              'payment_declined',
                              'payment_voided',
                              'payment_captured',
                              'payment_refunded'],
                  'dispute': ['dispute_canceled',
                              'dispute_evidence_required',
                              'dispute_expired',
                              'dispute_lost',
                              'dispute_resolved',
                              'dispute_won']}

try:
    response = api.workflows.update_workflow_condition('workflow_id', 'condition_id', request)
except CheckoutApiException as err:
    # API error
    request_id = err.request_id
    status_code = err.http_status_code
    error_details = err.error_details
    http_response = err.http_response
except CheckoutArgumentException as err:
    # Bad arguments

except CheckoutAuthorizationException as err:
    # Invalid authorization

File: nas_spec/code_samples/Node/workflows@{workflowId}@conditions@{workflowConditionId}/put.js
import { Checkout } from 'checkout-sdk-node';

let cko = new Checkout('your_client_secret_here', {
	client: 'ack_XXXXXXXXXXXX',
	scope: ['gateway'], // array of scopes
	environment: 'sandbox', // or "production"
});

// Or if you use api keys:
// const cko = new Checkout('sk_sbox_XXX', { pk: 'pk_sbox_XXX'}});

try {
	let workflows = await cko.workflows.updateCondition(
		'wf_2i7z3lwdoe5uzmomm7yzrytqdy',
		'wfc_ybu4t6aruwju5l6ymlc67ya5ne',
		{
			type: 'event',
			events: {
				gateway: ['card_verification_declined', 'card_verified', 'payment_approved'],
			},
		}
	);
} catch (err) {
	console.log(err.name);
}
